File: generated-code/codecraft/typescript/src/model/player.ts
import { StreamWrapper } from "../stream-wrapper";

export class Player {
    id: number
    score: number
    resource: number

    constructor(id: number, score: number, resource: number) {
        this.id = id;
        this.score = score;
        this.resource = resource;
    }

    static async readFrom(stream: StreamWrapper): Promise<Player> {
        let id;
        id = await stream.readInt();
        let score;
        score = await stream.readInt();
        let resource;
        resource = await stream.readInt();
        return new Player(id, score, resource)
    }

    async writeTo(stream: StreamWrapper) {
        let id = this.id;
        await stream.writeInt(id);
        let score = this.score;
        await stream.writeInt(score);
        let resource = this.resource;
        await stream.writeInt(resource);
    }
}

File: generated-code/codecraft/kotlin/src/main/kotlin/model/Player.kt
package model

import util.StreamUtil

class Player {
    var id: Int = 0
    var score: Int = 0
    var resource: Int = 0

    constructor(id: Int, score: Int, resource: Int) {
        this.id = id
        this.score = score
        this.resource = resource
    }

    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        StreamUtil.writeInt(stream, score)
        StreamUtil.writeInt(stream, resource)
    }

    override fun toString(): String {
        var stringBuilder = StringBuilder("Player { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("score: ")
        stringBuilder.append(score)
        stringBuilder.append(", ")
        stringBuilder.append("resource: ")
        stringBuilder.append(resource)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Player {
            var id: Int
            id = StreamUtil.readInt(stream)
            var score: Int
            score = StreamUtil.readInt(stream)
            var resource: Int
            resource = StreamUtil.readInt(stream)
            return Player(id, score, resource)
        }
    }
}